CONTRIBUTING.md

```diff
@@ -0,0 +1,114 @@
+# Contributing to HSR Optimizer
+
+We love your input! We want to make contributing to this project as easy and transparent as possible, whether it's:
+
+- Reporting a bug
+- Discussing the current state of the code
+- Submitting a fix
+- Proposing new features
+- Becoming a maintainer
+
+## We Develop with Github
+We use github to host code, to track issues and feature requests, as well as accept pull requests.
+
+## We Use [Github Flow](https://guides.github.com/introduction/flow/index.html), So All Code Changes Happen Through Pull Requests
+Pull requests are the best way to propose changes to the codebase. We actively welcome your pull requests:
+
+1. Fork the repo and create your branch from `beta`.
+2. If you've added code that should be tested, add tests.
+3. If you've changed APIs, update the documentation.
+4. Ensure the test suite passes.
+5. Make sure your code lints.
+6. Issue that pull request!
+
+## Report bugs using Github's [issues](https://github.com/fribbels/hsr-optimizer/issues)
+We use GitHub issues to track public bugs. Report a bug by [opening a new issue](https://github.com/fribbels/hsr-optimizer/issues/new); it's that easy!
+
+## Write bug reports with detail, background, and sample code
+Bugs should include reproduction steps, screenshots, and if possible, the metadata to reproduce the issue.  Exporting the HSR Optimizer database is a great start; attach it to your new issue!
+
+## Use a Consistent Coding Style
+* 2 spaces for indentation rather than tabs
+* You can try running `npm run lint` for style unification
+
+## Your first issue & pull request
+- Check current issues in the [BACKLOG](https://github.com/users/fribbels/projects/2)
+- Priority stack ranking and rough project status can be found in the [PRIORITY BOARD](https://github.com/users/fribbels/projects/2/views/2)
+- Find an issue to work on and set it status to `In Progress` by moving it from the Backlog column to the In Progress column.  *This lets other contributors know that the issue is being worked on!*
+
+After you have forked the repo and switched to the `beta` branch, create your working branch:
+```
+git checkout -b [feature|fix]/[your-branch-name]
+```
+
+## Commit Message Format
+Each commit message consists of a **header**, a **body**, and a **footer**. The header has a special format that includes a **type**, a **scope**, and a **subject**:
+
+***The header is mandatory and the scope of the header is optional.***
```

Comment:
I usually don't like those commit patterns. The only thing that a commit message should be is concise and contain all the information to be easier to find what was done when seeing git log/commit history.
We can however recommend it, but not make it mandatory.
But as we are using github flow reference, we can cite it directly:

```txt
Commit and push your changes to your branch. Give each commit a descriptive message to help you and future contributors understand what changes the commit contains. For example, fix typo or increase rate limit.
```

https://docs.github.com/en/get-started/using-github/github-flow#make-changes